src/tests/functional/plugin/shared/src/execution_graph_tests/fp16_compress_disable.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <tuple>
+#include <vector>
+#include <unordered_set>
+#include <string>
+#include <functional>
+
+#include <ie_core.hpp>
+#include <ngraph/function.hpp>
+#include <exec_graph_info.hpp>
+
+#include "common_test_utils/common_utils.hpp"
+#include "shared_test_classes/base/layer_test_utils.hpp"
+#include "functional_test_utils/skip_tests_config.hpp"
+#include "execution_graph_tests/fp16_compress_disable.hpp"
+#include "ie/ie_plugin_config.hpp"
+#include "transformations/rt_info/disable_fp16_compression.hpp"
+#include "openvino/runtime/system_conf.hpp"
+
+namespace ExecutionGraphTests {
+
+std::string ExecGraphDisableFP16Compress::getTestCaseName(testing::TestParamInfo<ExecGraphDisableFP16CompressSpecificParams> obj) {
+    std::ostringstream result;
+    bool matmulFP16Disabled;
+    bool FCFP16Disabled;
+    std::string targetDevice;
+    std::tie(matmulFP16Disabled, FCFP16Disabled, targetDevice) = obj.param;
+    result << "matmul_BF16_Disabled=" << matmulFP16Disabled << "_";
+    result << "FC_BF16_Disabled=" << FCFP16Disabled << "_";
+    result << "device=" << targetDevice;
+    return result.str();
+}
+
+void ExecGraphDisableFP16Compress::SetUp() {
+    std::tie(matmulFP16Disabled, FCFP16Disabled, targetDevice) =  this->GetParam();
+    create_model();
+}
+
+void ExecGraphDisableFP16Compress::TearDown() {
+}
+
+/*  test model:
+      ---------
+      |Input  |
+      ---------
+          |
+    -------------
+    | --------- |
+    | | Matmul| |
+    | --------- |
+    |     |     |
+    | --------- |
+    | |  FC   |
+    | --------- |
+    |-----------|
+          |
+      ---------
+      |Output |
+      ---------
+*/
+
+void ExecGraphDisableFP16Compress::create_model() {
+    auto A = std::make_shared<ov::op::v0::Parameter>(ov::element::f32, ov::PartialShape{4, 16});
+    auto B = std::make_shared<ov::op::v0::Parameter>(ov::element::f32, ov::Shape{16, 16});
+    ov::ParameterVector inputParams = {A , B};
+    auto matmul = std::make_shared<ov::op::v0::MatMul>(A, B);
+    matmul->set_friendly_name("Matmul0");
+    if (matmulFP16Disabled)
+        ov::disable_fp16_compression(matmul);
+    auto fcWeight = ngraph::builder::makeConstant(ov::element::f16, ov::Shape{16, 16}, std::vector<float>{0.0f}, true);
+    auto weightConvert = std::make_shared<ov::op::v0::Convert>(fcWeight, ov::element::f32);
+    auto fc = std::make_shared<ov::op::v0::MatMul>(matmul, weightConvert);
```

Comment:
FullyConnected is CPU plgin specific node, while you are implementing generic test. 
Why we need subgraph with 2 matmuls inside? Maybe single matmul will be enough?

Reply:
Applied. Only tested matmul with const weight.